lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -863,7 +862,28 @@ void OCLToSPIRVBase::transAtomicBuiltin(CallInst *CI,
           std::rotate(Args.begin() + 2, Args.begin() + OrderIdx,
                       Args.end() - Offset);
         }
-        return getSPIRVFuncName(OCLSPIRVBuiltinMap::map(Info.UniqName));
+        llvm::Type *AtomicBuiltinsReturnType =
+            CI->getCalledFunction()->getReturnType();
+        auto IsFPType = [](llvm::Type *ReturnType) {
+          return ReturnType->isHalfTy() || ReturnType->isFloatTy() ||
+                 ReturnType->isDoubleTy();
+        };
+        auto SPIRVFunctionName =
+            getSPIRVFuncName(OCLSPIRVBuiltinMap::map(Info.UniqName));
+        if (!IsFPType(AtomicBuiltinsReturnType))
+          return SPIRVFunctionName;
+        // Translate FP-typed atomic builtins. Currently we only need to
+        // translate atomic_fetch_[add, max, min] and atomic_fetch_[add, max,
+        // min]_explicit to related float instructions
+        auto SPIRFunctionNameForFloatAtomics =
+            llvm::StringSwitch<std::string>(SPIRVFunctionName)
+                .Case("__spirv_AtomicIAdd", "__spirv_AtomicFAddEXT")
+                .Case("__spirv_AtomicSMax", "__spirv_AtomicFMaxEXT")
+                .Case("__spirv_AtomicSMin", "__spirv_AtomicFMinEXT")
+                .Default("others");
```

Comment:
Hi, @AGindinson.
FP-typed atomic_fetch_add_explicit function have such IR as 
`float @_Z25atomic_fetch_add_explicitPU3AS1VU7_Atomicff12memory_order(float addrspace(1)* %a, float 0.000000e+00, i32 0)`.  Clang has no special dealt for the FP-typed atomic functions. 
And the FP-typed atomic_fetch_add_explicit function is dealt as function call instead of atomic builtins **before the patch**, see line 416 to 422 for details https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/lib/SPIRV/OCLToSPIRV.cpp#L416.
When add support for cl_ext_float_atomics, we need to translate the FP-typed atomic functions to proper SPIRV instructions, but SPIRVMap is always mapping ``atomic_fetch_add_explicit`` to ``__spirv_AtomicIAdd``, so I just add some extra dealt after SPIRVMap mapping actions.

Reply:
@haonanya, got it, thanks for the explanation. Personally, I'm fine with such an approach.